mlir/lib/Pass/PassRegistry.cpp

```diff
@@ -187,13 +211,15 @@ LogicalResult detail::pass_options::parseCommaSeparatedList(
   size_t nextElePos = findChar(optionStr, 0, ',');
   while (nextElePos != StringRef::npos) {
     // Process the portion before the comma.
-    if (failed(elementParseFn(optionStr.substr(0, nextElePos))))
+    if (failed(
+            elementParseFn(extractArgAndUpdateOptions(optionStr, nextElePos))))
       return failure();
 
-    optionStr = optionStr.substr(nextElePos + 1);
+    optionStr = optionStr.drop_front(); // drop the leading ','
```

Comment:
```suggestion
    // Drop the leading ','.
    optionStr = optionStr.drop_front();
```

Prefer non-trailing comments.